spark/v2.4/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteManifestsAction.java

```diff
@@ -478,6 +492,33 @@ public void testRewriteManifestsWithPredicate() throws IOException {
     Assert.assertEquals("Rows must match", expectedRecords, actualRecords);
   }
 
+  @Test
+  public void testRewriteManifestsNoOp() throws IOException {
```

Comment:
This is an additional test case for the scenario which was covered by spark procedure in 3.1 and above spark versions. 

Rest of the changes in the PR is same as #6695